src/data/move-attrs/turn-damaged-double-power-attr.ts

```diff
@@ -11,7 +11,7 @@ import { VariablePowerAttr } from "#app/data/move-attrs/variable-power-attr";
  */
 export class TurnDamagedDoublePowerAttr extends VariablePowerAttr {
   override apply(user: Pokemon, target: Pokemon, _move: Move, power: NumberHolder): boolean {
-    if (user.turnData.attacksReceived.find((r) => r.damage && r.sourceId === target.id)) {
+    if (user.turnData?.attacksReceived.find((r) => r.damage && r.sourceId === target.id)) {
```

Comment:
NIT: use `some` (I know this wasn't your change)
```suggestion
    if (user.turnData?.attacksReceived.some((r) => r.damage && r.sourceId === target.id)) {
```